clients/payment/src/components/PaymentForm.tsx

```diff
@@ -0,0 +1,369 @@
+      validate: validateRequired,
+      sanitize: value => value.trim(),
+    },
+  });
+
+  const onPressSubmit = () => {
+    submitForm(values => {
+      if (
+        hasDefinedKeys(values, [
+          "paymentMethod",
+          "iban",
+          "country",
+          "firstName",
+          "lastName",
+          "addressLine1",
+          "city",
+          "postalCode",
+          "state",
+        ])
+      ) {
+        console.log(values);
+      }
+    });
+  };
+
+  const languageOptions = useMemo(() => {
+    return languages.map(country => ({
+      name: country.native,
+      value: country.id,
+    }));
+  }, []);
+
+  return (
+    <ScrollView
+      style={{
+        backgroundColor: backgroundColor.default,
+      }}
+      contentContainerStyle={styles.container}
+    >
+      <Box direction="row" alignItems="center" justifyContent="spaceBetween">
+        <Box style={desktop ? styles.buttonItemDesktop : styles.buttonItem}>
+          <LakeButton
+            ariaLabel={t("common.cancel")}
+            icon="dismiss-regular"
+            mode="tertiary"
+            grow={true}
+            onPress={() => {}}
+          >
+            {t("common.cancel")}
+          </LakeButton>
+        </Box>
+
+        <Box
+          style={[desktop ? styles.buttonItemDesktop : styles.buttonItem, styles.selectLanguage]}
+        >
+          <LakeSelect
+            value={locale.language}
+            items={languageOptions}
+            hideErrors={true}
+            mode="borderless"
+            onValueChange={locale => {
+              setPreferredLanguage(locale);
+            }}
+          />
+        </Box>
+      </Box>
+
+      <Space height={24} />
+
+      {/* Just for tests */}
+      <SwanLogo
+        color={colors.swan[500]}
+        style={{
+          alignSelf: "center",
+          width: "144px",
+        }}
+      />
+
+      <Space height={24} />
+      <LakeText align="center">Merchant item</LakeText>
+      <Space height={12} />
+
+      <LakeHeading variant="h1" level={2} align="center">
+        {formatCurrency(Number(10), "EUR")}
+      </LakeHeading>
+
+      <Space height={32} />
+
+      <Field name="paymentMethod">
+        {({ value, onChange }) => (
+          <LakeLabel
+            style={desktop ? styles.segmentedControlDesktop : styles.segmentedControl}
+            label={t("paymentLink.paymentMethod")}
+            render={() => (
+              <SegmentedControl
+                mode="desktop"
+                selected={value}
+                items={items}
+                onValueChange={onChange}
+              />
+            )}
+          />
+        )}
+      </Field>
+
+      <Space height={24} />
+
+      <Field name="iban">
+        {({ value, valid, error, onChange }) => (
+          <LakeLabel
+            label={t("paymentLink.iban")}
+            render={() => (
+              <LakeTextInput value={value} valid={valid} error={error} onChangeText={onChange} />
```

Comment:
Don't forget to pass all data to inputs: `onBlur`, `ref` etc, so you have no surprises when starting validating them 🙂